packages/core/src/dialects/abstract/data-types.ts

```diff
@@ -1408,22 +1412,33 @@ export class DATE extends AbstractDataType<AcceptedDate> {
   constructor(precisionOrOptions?: number | DateOptions) {
     super();
 
-    this.options = {
-      precision: typeof precisionOrOptions === 'object' ? precisionOrOptions.precision : precisionOrOptions,
-    };
+    if (typeof precisionOrOptions === 'object') {
+      this.options = precisionOrOptions;
+    } else {
+      this.options = { precision: precisionOrOptions };
+    }
 
     if (this.options.precision != null && (this.options.precision < 0 || !Number.isInteger(this.options.precision))) {
       throw new TypeError('Option "precision" must be a positive integer');
     }
   }
 
+  protected _checkOptionSupport(dialect: AbstractDialect) {
+    super._checkOptionSupport(dialect);
+
+    if (!this.options.plain && !dialect.supports.dataTypes.DATETIME.offset) {
```

Comment:
I'd love to be able to be this strict but the non-plain behavior should not change for the time being

Reply:
I've removed this and replaced with a TODO